static/report/js/report.js

```diff
@@ -388,24 +363,23 @@
     
     function daysfilter() {
       matchesneeded++;
-      for (var d in daystoshow) {
-        if (daystoshow.hasOwnProperty(d)) {
-          var day = new Date(d).getDay();
-          if (day===0 && $('#rp_su').is(':checked')) { daystoshow[d]++; }
-          if (day===1 && $('#rp_mo').is(':checked')) { daystoshow[d]++; }
-          if (day===2 && $('#rp_tu').is(':checked')) { daystoshow[d]++; }
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1231252485/issues/source?bid=2452374&fileBranchId=2932983#l366)